tests/test_generate_classes.py

```diff
@@ -51,5 +52,7 @@ def test_weather_roundtrip(tmp_path):
             map={"key": "value", "another key": "another value"},
             enum="EnumA",
             record=Record(Field1="only field"),
+            circle=None
```

Comment:
Could you construct a second Weather instance use it here. I tried to and I faced an issue where the `fromdict` method cant construct the nested Weather record because it is typed as a `ForwardRef('Weather')` type which we don't have any handling for. We have to add some handling in `_handle_type` to catch that. It becomes a bit tricky to reconstruct the nested record because we only get the nested type as a string. One hacky way I imagine would be to pass the "parent" record type down and in cases of `ForwardRef` we would try to reconstruct it based on the passed down "parent". I think this should work since (currently) `ForwardRef` should only occur with cyclic schemas.